cmd/gaia/cli_test/cli_test.go

```diff
@@ -206,6 +170,35 @@ func TestGaiaCLISubmitProposal(t *testing.T) {
 	require.Equal(t, gov.VoteOptionToString(gov.OptionYes), vote.Option)
 }
 
+//___________________________________________________________________________________
+// helper methods
+
+func getTestingHomeDirs() (string, string) {
+	tmpDir := os.TempDir()
+	gaiadHome := fmt.Sprintf("%s%s.test_gaiad", tmpDir, string(os.PathSeparator))
+	gaiacliHome := fmt.Sprintf("%s%s.test_gaiacli", tmpDir, string(os.PathSeparator))
+	return gaiadHome, gaiacliHome
+}
+
+// returns flags, port, process
+func resetTestEnv(t *testing.T) (flags string, port string, proc *tests.Process) {
+	tests.ExecuteT(t, fmt.Sprintf("gaiad --home=%s unsafe_reset_all", gaiadHome))
+	executeWrite(t, fmt.Sprintf("gaiacli keys delete --home=%s foo", gaiacliHome), pass)
+	executeWrite(t, fmt.Sprintf("gaiacli keys delete --home=%s bar", gaiacliHome), pass)
+	chainID := executeInit(t, "gaiad init -o --name=foo --home="+gaiadHome)
+	executeWrite(t, fmt.Sprintf("gaiacli keys add --home=%s bar", gaiacliHome), pass)
+
+	// get a free port, also setup some common flags
+	servAddr, port, err := server.FreeTCPAddr()
+	require.NoError(t, err)
+	flags = fmt.Sprintf("--home=%s --node=%v --chain-id=%v", gaiacliHome, servAddr, chainID)
+
+	// start gaiad server
+	proc = tests.GoExecuteTWithStdout(t, fmt.Sprintf("gaiad start --home=%s --rpc.laddr=%v", gaiadHome, servAddr))
+
+	return flags, port, proc
+}
```

Comment:
Yo so although I like the idea of clumping some common functionality - I think we should be more verbose for these tests and expand this function BACK to the `TestXxx`s - The reason being is that these CLI tests have been designed to be readable like a script, I think it it's a step in the wrong direction abstracting anything out from the main test file which contains _CLI_ tests - somebody should be able to just look at the test line-by-line and replicate it's functionality it the CLI

Reply:
Will revert, thanks! Thats a good point I hadn't considered.